taxcalc/behavior.py

```diff
@@ -83,6 +90,72 @@ def update_behavior(self, revisions):
         self.set_year(precall_current_year)
         self._validate_parameter_values()
 
+    def update_behavior(self, reform):
+        # check that all revisions dictionary keys are integers
+        if not isinstance(reform, dict):
+            raise ValueError('ERROR: YYYY PARAM reform is not a dictionary')
+        if not reform:
+            return  # no reform to implement
+        reform_years = sorted(list(reform.keys()))
+        for year in reform_years:
+            if not isinstance(year, int):
+                msg = 'ERROR: {} KEY {}'
+                details = 'KEY in reform is not an integer calendar year'
+                raise ValueError(msg.format(year, details))
+        # check range of remaining reform_years
+        first_reform_year = min(reform_years)
+        if first_reform_year < self.start_year:
+            msg = 'ERROR: {} YEAR reform provision in YEAR < start_year={}'
+            raise ValueError(msg.format(first_reform_year, self.start_year))
+        if first_reform_year < self.current_year:
+            msg = 'ERROR: {} YEAR reform provision in YEAR < current_year={}'
+            raise ValueError(msg.format(first_reform_year, self.current_year))
+        last_reform_year = max(reform_years)
+        if last_reform_year > self.end_year:
+            msg = 'ERROR: {} YEAR reform provision in YEAR > end_year={}'
+            raise ValueError(msg.format(last_reform_year, self.end_year))
```

Comment:
All of the year validation logic may be overkill for the behavior class. However, I thought it would be interesting to show how much of the `Policy` code could be copied over.